backend/backend/services/business_logic.py

```diff
@@ -2,6 +2,44 @@
 
 from utils.logger import logger
 
+import json
+
+
+async def get_test_result(context: Context, subject: str):
+    logger.info(f"Getting test result for subject: {subject}")
+
+    adjacent_chain = context.chains.get_chains()['adjacent_queries_chain']
+    search_tool = context.tools.get_search_tool()
+    summary_chain = context.chains.get_chains()['summary_chain']
+
+    knowledge = []
+    for iter in range(2):
+        queries = await adjacent_chain.ainvoke(subject, 4,
+                                               knowledge="\n".join(knowledge))
+        responses = []
+        for query in queries:
+            artifact = await search_tool.ainvoke_tool_call_artifact(query)
+            response = {"query": query, "answer": artifact["answer"]}
+
+            if False:
```

Comment:
The `if False:` block on line 24 will never execute the summary generation code. This might be intentional, but if not, it should be reviewed or removed.